### PR TITLE
refactor: add JavaScript constrainer

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds the JS constrainer logic.

**Related issue(s)**
Blocked by #683

### CODE REVIEW

**Comment by jonaslagoni** [comment]
@magicmatatjahu there is gonna be quite a lot of duplicate logic between all the generators. However, they are only temporary until I get the larger picture of what can be shared.

File: src/generators/javascript/JavaScriptConstrainer.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import { ConstrainedAnyModel, ConstrainedBooleanModel, ConstrainedFloatModel, ConstrainedIntegerModel, ConstrainedMetaModel, ConstrainedObjectModel, ConstrainedReferenceModel, ConstrainedStringModel, ConstrainedTupleModel, ConstrainedTupleValueModel, ConstrainedArrayModel, ConstrainedUnionModel, ConstrainedEnumModel, ConstrainedDictionaryModel, ConstrainedEnumValueModel } from '../../models/ConstrainedMetaModel';
+import { AnyModel, BooleanModel, FloatModel, IntegerModel, MetaModel, ObjectModel, ReferenceModel, StringModel, TupleModel, ArrayModel, UnionModel, EnumModel, DictionaryModel } from '../../models/MetaModel';
+import { defaultPropertyKeyConstraints, PropertyKeyConstraintType } from './constrainer/PropertyKeyConstrainer';
+import { defaultModelNameConstraints, ModelNameConstraintType } from './constrainer/ModelNameConstrainer';
+import { defaultEnumKeyConstraints, EnumConstraintType } from './constrainer/EnumConstrainer';
+import { Logger } from '../../utils';
+
+export interface TypeScriptConstraints {
```
Comment (magicmatatjahu) [inline src/generators/javascript/JavaScriptConstrainer.ts:8]:
```suggestion
export interface JavaScriptConstraints {
```

**Review by magicmatatjahu — COMMENTED** [review]
Overall good, but you should change name of one interface :)

**Review by magicmatatjahu — APPROVED** [review]
LGTM!

**Comment by jonaslagoni** [comment]
@magicmatatjahu thanks!

/rtm

**Pull request merged**